mob

```diff
@@ -209,157 +226,201 @@ while not os.path.exists(os.path.join(top_level, ".mobconf")):
+    if git_commit:
+        docker_run.append('--env=GIT_COMMIT={}'.format(git_commit))
+
+    # Set up default ports to share with the Host OS
+    if not args.no_publish:
+        ports = [
+            "8080",
+            "8081",
+            "8443",
+            "3223",
+            "3225",
+            "3226",
+            "3228",
+            "4444",
+        ]
 
-jobs = get_cargo_build_jobs()
-if jobs is not None:
-    docker_run.extend(["--env", f"CARGO_BUILD_JOBS={jobs}"])
+    # debug options allow us to attach gdb when debugging failing tests
+    docker_run.append('--cap-add=SYS_PTRACE')
+
+elif args.image == 'signing-tools':
+    # place holder for now, Set up default ports to share with the Host OS
+    if not args.no_publish:
+        ports = []
+
+    # if we're running on linux and have an X11 environment set up, share the host network and xauth.
+    if platform.system().lower() == 'linux':
+        xauthority = os.environ.get('XAUTHORITY')
+        if xauthority:
+            docker_run.extend([
+                '--net=host',
+                '--env=DISPLAY',
+                '--env=XAUTHORITY',
+                '--volume={0}:{0}:ro'.format(xauthority)
+            ])
+
+else:
+    raise Exception('Unknown image')
 
-# If running interactively (with a tty), get a tty in the container also
-# This allows colored build logs when running locally without messing up logs in CI
+##
+# set interactive/tty if tty is available
+##
 if sys.stdout.isatty():
-    docker_run.extend(["-t"])
-
-ports = []
-# in prompt, use -i to get user input
-if args.action == "prompt":
-    docker_run.extend(["-i"])
-
-    # ports might be exposed to run clients or a local network
-    ports = [
-        "8080",
-        "8081",
-        "8443",
-        "3223",
-        "3225",
-        "3226",
-        "3228",
-        "4444",
-    ]
+    docker_run.append("-it")
 
-    # debug options allow us to attach gdb when debugging failing tests
-    docker_run.extend([
-        "--cap-add", "SYS_PTRACE",
-    ])
-
-ports.extend(args.expose or [])
+##
+# Add ports
+##
+if args.publish:
+    ports.extend(args.publish)
 for port in ports:
-    docker_run.extend(["--expose", port])
-
-for port in args.publish or []:
     docker_run.extend(["--publish", "{}:{}".format(port, port)])
 
-# Map in the ssh directory, or the ssh-agent socket
-if args.ssh_dir:
-    eprint("SSH Agent authentication disabled. You will need to run the following commands to build: ")
-    eprint('eval `ssh-agent`')
-    eprint('ssh-add /root/.ssh/<your-ssh-private-key>')
-    docker_run.extend(["--volume", os.path.expanduser("~/.ssh") + ":" + "/root/.ssh"])
-elif args.ssh_agent:
-    docker_run.extend(["--env", "SSH_AUTH_SOCK=/tmp/ssh_auth_sock"])
-    docker_run.extend(["--volume", os.environ["SSH_AUTH_SOCK"] + ":" + "/tmp/ssh_auth_sock"])
+##
+# Add extra environment variables.
+##
+if args.env:
+    for e in args.env:
+        docker_run.append('--env={}'.format(e))
```

Comment:
IIUC this `if` is unnecessary
```suggestion
for e in args.env:
    docker_run.append('--env={}'.format(e))
```

Reply:
We need an if here because arg.env isn't iterable unless you specify an option.

```
Traceback (most recent call last):
  File "/home/jgreat/git/mcf/mobilecoin/./mob", line 359, in <module>
    for e in args.env:
TypeError: 'NoneType' object is not iterable
```